model/assets.go

```diff
@@ -82,12 +81,12 @@ func (c AssetConverter) FromString(s string) (Asset, error) {
 }
 
 // MustFromString converts from a string to an asset, failing on errors
-func (c AssetConverter) MustFromString(s string) Asset {
+func (c AssetConverter) MustFromString(s string) (Asset, error) {
 	a, e := c.FromString(s)
 	if e != nil {
-		log.Fatal(fmt.Errorf("exiting on an error-enforced asset conversion: %s", e))
+		return "", fmt.Errorf("exiting on an error-enforced asset conversion: %s", e)
```

Comment:
this method should be a fatal -- MustFromString forces you to panic by definition.
let's not make any conversions from fatal to returning an error in this PR. 

the logger should be injected into the AssetConverter. Let's try that for now and see how big the change is

Reply:
ok